model/postgres/postgres_resource.rb

```diff
@@ -43,7 +43,11 @@ def path
   def display_state
     return "deleting" if destroy_set? || strand.nil? || strand.label == "destroy"
     return "unavailable" if representative_server&.strand&.label == "unavailable"
-    return "converging" if strand.children.any? { it.prog == "Postgres::ConvergePostgresResource" }
+    if strand.children.any? { it.prog == "Postgres::ConvergePostgresResource" }
+      return "converging" if has_enough_ready_servers? && in_maintenance_window?
+      return "waiting maintenance window" if has_enough_ready_servers? && !in_maintenance_window?
```

Comment:
nit, not sure if the UI would look okay with this though
```suggestion
      return "waiting for maintenance window" if has_enough_ready_servers? && !in_maintenance_window?
```